### PR TITLE
Tech/285/map tree view component test coverage

### PR DESCRIPTION
**Pull request by FernandoES**
Added Test coverage to mapTreeViewLevelComponent.ts
Related to #285
## Description

{Descriptive pull request text}

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

* [ ] **All** requirements mentioned in the issue are implemented
* [ ] Does match the Code of Conduct and the Contribution file 
* [ ] Task has its own **GitHub issue** (something it is solving)
  * [ ] Issue number is **included in the commit messages** for traceability
* [ ] **Update the README.md** with any changes/additions made
* [ ] **Update the CHANGELOG.md** with any changes/additions made
* [ ] **Enough test coverage to ensure that uncovered changes do not break functionality**
* [ ] **All tests pass**    
* [ ] **Descriptive pull request text**, answering:
  + What problem/issue are you fixing?
  + What does this PR implement and how? 
* [ ] **Assign your PR to someone** for a code review
  + This person _will be contacted **first**_ if a bug is introduced into `master`
* [ ] **Manual testing** did not fail

### CODE REVIEW

File: visualization/app/codeCharta/ui/mapTreeView/mapTreeViewLevelComponent.spec.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            let subscriber: MapTreeViewHoverEventSubscriber = {
+                onShouldHoverNode: jest.fn(),
+                onShouldUnhoverNode: jest.fn()
+            };
+
+            MapTreeViewLevelController.subscribeToHoverEvents($rootScope, subscriber);
+
+            expect($rootScope.$on).toBeCalledWith("should-hover-node", expect.any(Function));
+            expect($rootScope.$on).toBeCalledWith("should-unhover-node", expect.any(Function));
+            expect(subscriber.onShouldHoverNode).toBeCalled;
```
Comment (alschmut) [inline visualization/app/codeCharta/ui/mapTreeView/mapTreeViewLevelComponent.spec.ts]: These two statements always return true. It should actually be like:
`expect(subscriber.onShouldHoverNode).toBeCalled();` but these unfortunately always return false. Any ideas?

Comment (FernandoES) [reply visualization/app/codeCharta/ui/mapTreeView/mapTreeViewLevelComponent.spec.ts]: You are right. I add a function called "whatever" to the subscriber mock and it returns true.
I think it has something to do with being inside from a anonymous function or being given as an argument into a function.
I added the line

            expect($rootScope.$on).toBeCalled();
and returns true, so toBeCalled() works

Comment (FernandoES) [reply visualization/app/codeCharta/ui/mapTreeView/mapTreeViewLevelComponent.spec.ts]: My opinion: The problem is that subscriber is an interface.

Comment (FernandoES) [reply visualization/app/codeCharta/ui/mapTreeView/mapTreeViewLevelComponent.spec.ts]: I completely changed the perspective

File: visualization/app/codeCharta/ui/mapTreeView/mapTreeViewLevelComponent.spec.ts [hunk 2]
```diff
@@ -197,17 +210,22 @@ describe("MapTreeViewLevelController", () => {
...
             expect($rootScope.$on).toBeCalledWith("should-unhover-node", expect.any(Function));
-            expect(subscriber.onShouldHoverNode).toBeCalled;
-            expect(subscriber.onShouldUnhoverNode).toBeCalled;
+
+            $rootScope.$on("should-hover-node", (event: IAngularEvent, node: CodeMapNode) => {
+                expect(node).toBe(mapTreeViewLevelController.node);
+            });
+
+            $rootScope.$on("should-unhover-node", (event: IAngularEvent, node: CodeMapNode) => {
+                expect(node).toBe(mapTreeViewLevelController.node);
```
Comment (alschmut) [inline visualization/app/codeCharta/ui/mapTreeView/mapTreeViewLevelComponent.spec.ts]: Unfortunately the test still passes with random values like:
`expect(node).toBe("whatever");`

File: visualization/app/codeCharta/ui/mapTreeView/mapTreeViewLevelComponent.spec.ts [hunk 3]
```diff
@@ -211,21 +213,25 @@ describe("MapTreeViewLevelController", () => {
 
         it("Subscribe Hover", () => {
 
+            instantiateModule("app.codeCharta.ui.mapTreeView");
```
Comment (mrpatpat) [inline visualization/app/codeCharta/ui/mapTreeView/mapTreeViewLevelComponent.spec.ts]: @alschmut @FernandoES  the problem was that in order to test things that rely on the event system of angular you need to use the "real" $rootScope or a mock that handles the events correctly. As long as $rootScope is just a "dumb" mock with no real event system implementation it cannot broadcast events. 
The solution here is to instatiate the module and get the real rootScope:)

**Pull request merged**